### ISSUE TITLE
sanjo_long-running-child-process error

### ISSUE THREAD

**Issue by AdamBrodzinski** [xolvio/meteor-jasmine#279]
I've been having issues getting Jasmine running and I can't figure out why. I tried to make a new app to double check and am getting the same output. 

Server unit does work. All the rest fail. It seems like the lib folder isn't there and also the path is different if you compare the error with the tree output:

```text
sanjo_long-running-child-process|⇒ cd /Users/adam/projects/sandbox/sanjotest/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process
sanjo_long-running-child-process|⇒ tree
.
└── packages
    └── sanjo_long-running-child-process
        └── lib
            └── spawnScript.js

3 directories, 1 file
````

This happens if I drop a file in client or server integration folders:

```text
W20150908-21:21:26.095(-4)? (STDERR) [sanjo:jasmine]: { [Error: ENOENT, no such file or directory '/Users/adam/projects/meteor-webpack-react/meteor_core/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js']
W20150908-21:21:26.095(-4)? (STDERR)   errno: 34,
W20150908-21:21:26.095(-4)? (STDERR)   code: 'ENOENT',
W20150908-21:21:26.095(-4)? (STDERR)   path: '/Users/adam/projects/meteor-webpack-react/meteor_core/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js',
W20150908-21:21:26.095(-4)? (STDERR)   syscall: 'chmod' }
W20150908-21:21:26.097(-4)? (STDERR) [sanjo:jasmine]: { [Error: ENOENT, no such file or directory '/Users/adam/projects/meteor-webpack-react/meteor_core/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js']
W20150908-21:21:26.097(-4)? (STDERR)   errno: 34,
W20150908-21:21:26.097(-4)? (STDERR)   code: 'ENOENT',
W20150908-21:21:26.098(-4)? (STDERR)   path: '/Users/adam/projects/meteor-webpack-react/meteor_core/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js',
W20150908-21:21:26.098(-4)? (STDERR)   syscall: 'chmod' }
^C
```

and also if I drop into client unit I get:
```text
=> App running at: http://localhost:3000/
I20150908-21:32:42.668(-4)? Exception in queued task: Error: ENOENT, no such file or directory '/Users/adam/projects/sandbox/sanjotest/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js'
I20150908-21:32:42.668(-4)?     at Object.fs.chmodSync (fs.js:833:18)
I20150908-21:32:42.668(-4)?     at LongRunningChildProcess.sanjo.LongRunningChildProcess.LongRunningChildProcess.spawn (packages/sanjo_long-running-child-process/packages/sanjo_long-running-child-process.js:288:8)
I20150908-21:32:42.668(-4)?     at Object.KarmaInternals.startKarmaServer (packages/sanjo:karma/main.js:73:1)
I20150908-21:32:42.669(-4)?     at Object.Karma.start (packages/sanjo:karma/main.js:13:1)
I20150908-21:32:42.669(-4)?     at [object Object]._.extend.startKarma (packages/sanjo:jasmine/src/client/unit/ClientUnitTestFramework.js:54:1)
I20150908-21:32:42.669(-4)?     at Object.testsCursor.observe.added (packages/sanjo:jasmine/src/lib/lazyStart.js:21:1)
I20150908-21:32:42.669(-4)?     at Object.added (packages/underscore/underscore.js:750:1)
I20150908-21:32:42.669(-4)?     at [object Object].observeChangesCallbacks.added (packages/minimongo/observe.js:153:1)
I20150908-21:32:42.669(-4)?     at self.applyChange.added (packages/minimongo/observe.js:53:1)
I20150908-21:32:42.669(-4)?     at packages/mongo/observe_multiplex.js:183:1
I20150908-21:32:42.669(-4)?     at Array.forEach (native)
I20150908-21:32:42.669(-4)?     at Function._.each._.forEach (packages/underscore/underscore.js:105:1)
I20150908-21:32:42.669(-4)?     at Object.task (packages/mongo/observe_multiplex.js:177:1)
I20150908-21:32:42.669(-4)?     at [object Object]._.extend._run (packages/meteor/fiber_helpers.js:147:1)
I20150908-21:32:42.670(-4)?     at packages/meteor/fiber_helpers.js:125:1
W20150908-21:32:42.802(-4)? (STDERR) connections property is deprecated. Use getConnections() method
```

**Comment by Sanjo** [comment]
Are you using the 1.2 release candidate?

**Comment by AdamBrodzinski** [comment]
No, was using `METEOR@1.1.0.3`. 

I just tried to update to RC-7 and ran it but it blows up with a different error (also the html-reporter overtakes the app full screen using that):

```
 Could not mock the export NpmModuleMongodb of the package npm-mongo. Will continue anyway. [TypeError: Cannot call method 'substring' of undefined] TypeError: Cannot call method 'substring' of undefined
W20150908-23:05:13.297(-4)? (STDERR)     at ObjectID.Object.defineProperty.get [as generationTime] (/Users/adam/.meteor/packages/npm-mongo/.1.4.39-rc.0_1.1q1f0op++os+web.browser+web.cordova/npm/node_modules/mongodb/node_modules/bson/lib/bson/objectid.js:259:55)
```

Thanks!

**Comment by tobias-f** [comment]
Hi.

I am having the same issue. I am using #281 METEOR@1.2.0.1

    Could not mock the export NpmModuleMongodb of the package npm-mongo. Will continue anyway. [TypeError: Cannot call method 'substring' of undefined] TypeError: Cannot call method 'substring' of undefined

Any leads on that?

Thanks!

**Comment by think01** [comment]
Same for me (just updated to meteor 1.2). I noticed I'm using sanjo:karma 1.7 while there is available version 2.0, but meteor refuses to do the update. Can be related to this issue?

**Comment by Sanjo** [comment]
I will release a new version in the next 60 minutes.

**Comment by think01** [comment]
Thank you so much! Great work!
I still see the update advice about sanjo:karma:

    Newer versions of the following indirect dependencies are available:
     * sanjo:karma 1.7.0 (2.0.0 is available)
Does it mean I have to force update that package? Issuing "meteor update" does nothing.

**Comment by Sanjo** [comment]
@think01 No, the current version depends on sanjo:karma@1.7.0. The develop version depends on 2.0.0, but it is not ready for release yet.

**Comment by think01** [comment]
I see. Thank you again, now the thread issue is gone for me.

**Comment by seanfcarroll** [comment]
I had the same issue, and have now updated to the (official) Meteor 1.2.0.1, and this resolves the issue.

**Comment by derwaldgeist** [comment]
Having the same problem with Meteor 1.1.0.3. Is there anything I can do if I do not want to update to Meteor 1.2 for now?

**Comment by Sanjo** [comment]
The `NpmModuleMongodb` error is fixed in sanjo:jasmine@0.19.x.

The error `Exception in queued task: Error: ENOENT, no such file or directory '/Users/adam/projects/sandbox/sanjotest/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js'` seems to me that it is a meteor tool error. Try `meteor update outside of a project`. If that does not help, please create a new issue. I need your operation system, the version of sanjo:jasmine, velocity:core, sanjo:long-running-child-process and the meteor tool that you use. The info is in the .meteor/versions file and the output of `meteor show meteor-tool`.

**Comment by elijahlofgren** [comment]
In case this helps anyone, I didn't want to upgrade to Meteor 1.2 just yet and simply cloning https://github.com/Sanjo/meteor-long-running-child-process/ into my apps "packages" directory fix the "Exception in callback of async function: Error: ENOENT, no such file or directory '[MY_APP]/.meteor/local/build/programs/server/assets/packages/sanjo_long-running-child-process/lib/spawnScript.js'" error I was running into.
